lib/iris/coords.py

```diff
@@ -1948,24 +1948,38 @@ def collapsed(self, dims_to_collapse=None):
         if np.issubdtype(self.dtype, np.str_):
             # Collapse the coordinate by serializing the points and
             # bounds as strings.
-            def serialize(x):
-                return "|".join([str(i) for i in x.flatten()])
+            def serialize(x, axis):
+                if axis is None or len(axis) == x.ndim:
```

Comment:
I'm not 100% sure about this `len(axis) == x.ndim`.  Is there any way there could be repeated values in `axis`?

Reply:
Removed.